packages/graph/hash_graph/lib/graph/src/store/postgres/knowledge/link/read.rs

```diff
@@ -3,118 +3,163 @@ use error_stack::{IntoReport, Report, Result, ResultExt};
-        let link = self
-        .client
-        .as_client()
+async fn by_link_type_by_source_entity_id(
+    client: &(impl GenericClient + Sync),
+    link_type_uri: &BaseUri,
+    link_type_version: u32,
+    source_entity_id: EntityId,
+) -> Result<Vec<Links>, QueryError> {
+    let link =
+        client
         .query_one(
             r#"
-                -- Gather all single-links
-                WITH single_links AS (
-                    SELECT link_type_version_id, target_entity_id
-                    FROM links
-                    INNER JOIN ids ON ids.version_id = links.link_type_version_id
-                    WHERE active AND NOT multi AND source_entity_id = $1 AND base_uri = $2 AND "version" = $3
-                ),
-                -- Gather all multi-links
-                multi_links AS (
-                    SELECT link_type_version_id, ARRAY_AGG(target_entity_id ORDER BY multi_order ASC) AS target_entity_ids
-                    FROM links
-                    INNER JOIN ids ON ids.version_id = links.link_type_version_id
-                    WHERE active AND multi AND source_entity_id = $1 AND base_uri = $2 AND "version" = $3
-                    GROUP BY link_type_version_id
-                )
-                -- Combine single and multi links with null values in rows where the other doesn't exist
-                SELECT link_type_version_id, target_entity_id AS single_link, NULL AS multi_link FROM single_links 
-                UNION 
-                SELECT link_type_version_id, NULL AS single_link, target_entity_ids AS multi_link from multi_links
-                "#,
-            &[&source_entity_id, link_type_uri.base_uri(), &version],
+            -- Gather all single-links
+            WITH single_links AS (
+                SELECT link_type_version_id, target_entity_id
+                FROM links
+                INNER JOIN ids ON ids.version_id = links.link_type_version_id
+                WHERE active AND NOT multi AND source_entity_id = $1 AND base_uri = $2 AND "version" = $3
+            ),
+            -- Gather all multi-links
+            multi_links AS (
+                SELECT link_type_version_id, ARRAY_AGG(target_entity_id ORDER BY multi_order ASC) AS target_entity_ids
+                FROM links
+                INNER JOIN ids ON ids.version_id = links.link_type_version_id
+                WHERE active AND multi AND source_entity_id = $1 AND base_uri = $2 AND "version" = $3 GROUP BY link_type_version_id
+            )
+            -- Combine single and multi links with null values in rows where the other doesn't exist
+            SELECT link_type_version_id, target_entity_id AS single_link, NULL AS multi_link
+            FROM single_links
+            UNION
+            SELECT link_type_version_id, NULL AS single_link, target_entity_ids AS multi_link
+            FROM multi_links
+            "#,
+            &[&source_entity_id, link_type_uri, &i64::from(link_type_version)],
```

Comment:
Here `parameter_list` can be utilized as well

Reply:
Not possible here because it is using `query_one` rather than `query_raw`